pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -240,26 +314,147 @@ def _xyx_decomposition(U, wire, return_global_phase=False):
 
     # Choose gamma such that exp(-i*gamma)*U is special unitary (detU==1).
     U = math.expand_dims(U, axis=0) if len(U.shape) == 2 else U
-    U_det1, gamma = _convert_to_su2(U, return_global_phase=True)
+    U_det1, gammas = _convert_to_su2(U, return_global_phase=True)
 
     # Compute \phi, \theta and \lambda after analytically solving for them from
     # U_det1 = expm(1j*\phi*PauliX) expm(1j*\theta*PauliY) expm(1j*\lambda*PauliX)
-    lam_plus_phi = math.arctan2(-math.imag(U_det1[:, 0, 1]), math.real(U_det1[:, 0, 0]) + EPS)
-    lam_minus_phi = math.arctan2(math.imag(U_det1[:, 0, 0]), -math.real(U_det1[:, 0, 1]) + EPS)
-    lam = lam_plus_phi + lam_minus_phi
-    phi = lam_plus_phi - lam_minus_phi
+    lams_plus_phis = math.arctan2(-math.imag(U_det1[:, 0, 1]), math.real(U_det1[:, 0, 0]) + EPS)
+    lams_minus_phis = math.arctan2(math.imag(U_det1[:, 0, 0]), -math.real(U_det1[:, 0, 1]) + EPS)
+    lams = lams_plus_phis + lams_minus_phis
+    phis = lams_plus_phis - lams_minus_phis
+
+    # Wrap angles to range [-\pi, \pi]
+    lams = (lams + qml.numpy.pi) % (2 * qml.numpy.pi) - qml.numpy.pi
+    phis = (phis + qml.numpy.pi) % (2 * qml.numpy.pi) - qml.numpy.pi
 
     # The following conditional attempts to avoid 0 / 0 errors. Either the
     # sine is 0 or the cosine, but not both.
-    if math.allclose(lam_plus_phi, 0):
-        theta = 2 * math.arccos(math.real(U_det1[:, 1, 1]) / (math.cos(lam_plus_phi) + EPS))
-    else:
-        theta = 2 * math.arccos(-math.imag(U_det1[:, 0, 1]) / (math.sin(lam_plus_phi) + EPS))
+    thetas = math.array(
```

Comment:
Not required, but one could solve this with a `math.where` call, I think :)

Reply:
I didn't know about this function, I'll use it.